sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -815,6 +815,24 @@ class Dataset[T] private[sql](
     println(showString(numRows, truncate, vertical))
   // scalastyle:on println
 
+  /**
+   * Returns the default number of rows to show when the show function is called without
+   * a user specified max number of rows.
+   * @since 2.3.0
+   */
+  private def numberOfRowsToShow(): Int = {
+    this.sparkSession.conf.get("spark.sql.show.defaultNumRows", "20").toInt
```

Comment:
I would add it to `SQLConf`

Reply:
+1

Reply:
+1

Reply:
How about `spark.sql.defaultNumRowsInShow`?

Reply:
will do